raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -3274,6 +3320,37 @@ private long pollFollowerAsVoter(FollowerState state, long currentTimeMs) {
     private long pollFollowerAsObserver(FollowerState state, long currentTimeMs) {
         if (state.hasFetchTimeoutExpired(currentTimeMs)) {
             return maybeSendFetchToAnyBootstrap(currentTimeMs);
+        } else if (partitionState.lastKraftVersion().isReconfigSupported() && followersAlwaysFlush &&
+            quorumConfig.autoJoinEnable() && state.hasAddRemoveVoterPeriodExpired(currentTimeMs)) {
```

Comment:
Okay. I think we should document why we require both `followersAlwaysFlush` and `autoJoinEnable` to be true.